src/librustc_typeck/diagnostics.rs

```diff
@@ -1972,6 +1972,89 @@ To learn more about traits, take a look at the Book:
 https://doc.rust-lang.org/book/traits.html
 "##,
 
+E0174: r##"
+This error occurs because of the explicit use of unboxed closure methods
+that are an experimental feature in current Rust version.
+
+Example of erroneous code:
+
+```compile_fail
+fn foo<F: Fn(&str)>(mut f: F) {
+    f.call(("call",));
+    // error: explicit use of unboxed closure method `call`
+    f.call_mut(("call_mut",));
+    // error: explicit use of unboxed closure method `call_mut`
+    f.call_once(("call_once",));
+    // error: explicit use of unboxed closure method `call_once`
+}
+
+fn bar(name: &str) {
+    println!("Calling {} it works!", name);
+}
+
+fn main() {
+    foo(bar);
+}
+```
+
+Rust’s implementation of closures is a bit different than other languages.
```

Comment:
It comes from the documentation take a look:
https://doc.rust-lang.org/book/closures.html#closure-implementation